test/common.py

```diff
@@ -82,6 +83,7 @@
 
 TEST_ROOT = path_from_root('test')
 LAST_TEST = path_from_root('out/last_test.txt')
+PREVIOUS_TEST_RUN_RESULTS_FILE = path_from_root('out/previous_test_run_results.json')
```

Comment:
What happens when we just this same file to write results from different test suites?  I guess the names of the specific tests won't match so results from a different test suite will be mostly ignored?

Reply:
The JSON file has sections for different test suites (`core1.test_atexit` vs `core2.test_atexit`) but then also `test_atexit` without distinction of the test suite.

The first accumulates results from specific suite, and the general one across all suites.

This way if a test fails in one suite, it gets bumped in fail frequency in other suites runs.